analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd

```diff
@@ -0,0 +1,290 @@
+This subtype harbors an H3F3A K28M or G35R/V mutation
+```{r}
+h3_biospecimen <- lgat_snv %>%
+  filter(H3.1_mut =="Yes"| H3.2_mut =="Yes" |H3.3_mut =="Yes") %>%
+  mutate(molecular_subtype = "LGG, H3") %>%
+  select("Kids_First_Biospecimen_ID","molecular_subtype") 
+
+```
+ 
+### LGG, MYB/MYBL1
+
+This subtype harbors either a MYB-QKI fusion or other MYB or MYBL1 fusion
+```{r}
+myb_biospecimen <- lgat_fus %>%
+  filter(MYB_fus=="Yes") %>%
+  mutate(molecular_subtype = "LGG, MYB/MYBL1") %>%
+  select("Kids_First_Biospecimen_ID","molecular_subtype")
+```
+
+### LGG, CDKN2A/B
+
+This subtype harbors focal CDKN2A and/or CDKN2B deletion
+This is a secondary co-occurring alteration with prognostic significance.
+```{r}
+
+cdkn_biospecimen <- lgat_cnv %>%
+  filter(CDKN2A_DEL=="Yes"| CDKN2B_DEL == "Yes") %>%
+  mutate(molecular_subtype = "LGG, CDKN2A/B") %>%
+  select("Kids_First_Biospecimen_ID","molecular_subtype") 
+```
+
+### Compile
+```{r}
+
+all_dna_subtype <- bind_rows(nf1_biospecimen,
+                         brafV600_biospecimen,
+                         otherMAPK_snv_biospecimen,
+                         rtk_snv_biospecimen,
+                         fgfr_cnv_biospecimen,
+                         fgfr_snv_biospecimen,
+                         idh_biospecimen,
+                         h3_biospecimen,
+                         cdkn_biospecimen
+) %>%
+  rename(Kids_First_Biospecimen_ID_DNA=Kids_First_Biospecimen_ID) %>%
+  # add Kids_First_Biospecimen_ID_RNA to dna subtypes dataframe
+  left_join(all_lgat_matched,by="Kids_First_Biospecimen_ID_DNA") 
+
+all_rna_subtype <- bind_rows(kiaabraf_biospecimen,
+                             otherMAPK_fus_biospecimen,
+                             rtk_fus_biospecimen,
+                             fgfr_fus_biospecimen,
+                             myb_biospecimen) %>%
+  rename(Kids_First_Biospecimen_ID_RNA=Kids_First_Biospecimen_ID) %>%
+  # add Kids_First_Biospecimen_ID_DNA to rna subtypes dataframe
+  left_join(all_lgat_matched,by="Kids_First_Biospecimen_ID_RNA")
+
+  
+all_subtype <- bind_rows(all_rna_subtype,
+                         all_dna_subtype) %>%
+  full_join(all_lgat_matched, by=c("Kids_First_Biospecimen_ID_RNA","Kids_First_Biospecimen_ID_DNA",
+                                   "sample_id",
+                                   "Kids_First_Participant_ID")) %>%
+  # lets group because there are some instances where LGAT DNA/RNA have multiple subtypes
+  group_by(sample_id,Kids_First_Participant_ID,Kids_First_Biospecimen_ID_RNA,Kids_First_Biospecimen_ID_DNA) %>%
+  # merging them here with summarise
+  summarise(molecular_subtype= toString(molecular_subtype)) %>%
+  # formatting molecular_subtype columns
+  mutate( molecular_subtype = case_when(
+    # multiple subtypes are summarised 
+    # with a redudant ",LGG," from instances 
+    # like molecular_subtype ==LGG, BRAF V600E, LGG, CDKN2A/B
+    grepl(", LGG,",molecular_subtype ) ~ 
+      gsub(", LGG,",",",molecular_subtype ),
```

Comment:
Would you be better off not including `LGG, ` in any of the steps above? Instead you could use `tidyr::unite()` to combine the columns separated by `, ` and then I think you could use `case_when()` with something like (untested):

```R
case_when(
  molecular_subtype == "NA" ~ "LGG, To be classified",
  TRUE ~ stringr::str_c("LGG, ", molecular_subtype)
)
```

Reply:
I tried the unite(all_subtype,"molecular_subtype",sep=",") and it collapsed all columns in `all_subtype` it didn't look right, am I missing something? unite() seems to combine col, which seems like it's not applicable here since we have 1 molecular_subtype column.

I used summarise since we need to combine molecular_subtype per unique row of ids (ie. combine if there are 2 rows 1 from all_dna_subtype and another 1 from all_rna_subtype only differing on molecular_subtype)

Reply:
I did implement your suggestion in your comment, that was useful to clean up some code